File: app/src/main/java/dev/ragnarok/fenrir/fragment/search/criteria/AudioPlaylistSearchCriteria.java
package dev.ragnarok.fenrir.fragment.search.criteria;

import android.os.Parcel;

public final class AudioPlaylistSearchCriteria extends BaseSearchCriteria {

    public static final Creator<AudioPlaylistSearchCriteria> CREATOR = new Creator<AudioPlaylistSearchCriteria>() {
        @Override
        public AudioPlaylistSearchCriteria createFromParcel(Parcel in) {
            return new AudioPlaylistSearchCriteria(in);
        }

        @Override
        public AudioPlaylistSearchCriteria[] newArray(int size) {
            return new AudioPlaylistSearchCriteria[size];
        }
    };

    public AudioPlaylistSearchCriteria(String query) {
        super(query);
    }

    private AudioPlaylistSearchCriteria(Parcel in) {
        super(in);
    }

    @Override
    public int describeContents() {
        return 0;
    }
}

File: app/src/main/java/dev/ragnarok/fenrir/fragment/search/criteria/AudioPlaylistSearchCriteria.kt
package dev.ragnarok.fenrir.fragment.search.criteria

import android.os.Parcel
import android.os.Parcelable

class AudioPlaylistSearchCriteria : BaseSearchCriteria {
    constructor(query: String?) : super(query)
    private constructor(`in`: Parcel) : super(`in`)

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<AudioPlaylistSearchCriteria> =
            object : Parcelable.Creator<AudioPlaylistSearchCriteria> {
                override fun createFromParcel(`in`: Parcel): AudioPlaylistSearchCriteria {
                    return AudioPlaylistSearchCriteria(`in`)
                }

                override fun newArray(size: Int): Array<AudioPlaylistSearchCriteria?> {
                    return arrayOfNulls(size)
                }
            }
    }
}